pilot/pkg/config/coredatamodel/discovery.go

```diff
@@ -81,132 +150,138 @@ func (d *MCPDiscovery) Services() ([]*model.Service, error) {
 
 // GetProxyServiceInstances returns service instances co-located with a given proxy
 func (d *MCPDiscovery) GetProxyServiceInstances(proxy *model.Proxy) ([]*model.ServiceInstance, error) {
+	if err := d.initializeCache(); err != nil {
+		return nil, err
+	}
 	out := make([]*model.ServiceInstance, 0)
 
 	// There is only one IP for kube registry
 	proxyIP := proxy.IPAddresses[0]
 
-	// add not ready endpoint
-	d.notReadyEndpointsMu.Lock()
-	defer d.notReadyEndpointsMu.Unlock()
-	for addr, conf := range d.notReadyEndpoints {
-		notReadyIP, port, err := net.SplitHostPort(addr)
-		if err != nil {
-			continue
-		}
-		notReadyPort, err := strconv.Atoi(port)
-		if proxyIP != notReadyIP || err != nil {
-			continue
-		}
-		se, ok := conf.Spec.(*networking.ServiceEntry)
-		if !ok {
-			return nil, errors.New("getProxyServiceInstances: wrong type")
-		}
-		for _, h := range se.Hosts {
-			for _, svcPort := range se.Ports {
-				out = append(out, d.notReadyServiceInstance(se, svcPort, conf, h, notReadyIP, notReadyPort))
+	d.cacheMutex.Lock()
+	defer d.cacheMutex.Unlock()
+
+	// svcInstances by a given namespace
+	if proxy.ConfigNamespace != model.NamespaceAll {
```

Comment:
ConfigNamespace != "", empty means not specified, this can happen on non k8s env

Reply:
I think for now this is serving galley which would be only k8s, but that is a good point we should consider this when we are serving more config sources that are non k8s environments.